src/ert/_c_wrappers/enkf/enkf_fs_manager.py

```diff
@@ -102,10 +102,16 @@ def __getitem__(self, case_name: str) -> EnkfFs:
         if case_name in self.open_storages:
             return self.open_storages[case_name]
         elif case_name in self.cases:
-            return self.add_case(case_name)
+            file_system = EnkfFs.createFileSystem(
+                self.storage_path / case_name,
+                self._ensemble_config,
+                self._ensemble_size,
+                self.read_only,
+            )
+            self._add_to_open(file_system)
+            return file_system
         else:
             raise KeyError(f"No such case name: {case_name} in {self.cases}")
 
-    def umount(self) -> None:
-        while len(self.open_storages) > 0:
-            self._drop_oldest_file_system()
+    def __iter__(self) -> str:
+        yield from self.cases
```

Comment:
Is this ever used?

Also, if one would want to use it they would have to look up the case after retrieving it. e.g.:

```
for case_name in storage_manager:
    storage_manager[case_name].some_action
```

Might be alright, just wasn't super obvious what one would get by iterating it..